server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeShutdownAllocationDecider.java

```diff
@@ -27,16 +27,22 @@ public class NodeShutdownAllocationDecider extends AllocationDecider {
 
     private static final String NAME = "node_shutdown";
 
+    private static final Decision YES_EMPTY_SHUTDOWN_METADATA = Decision.single(Decision.Type.YES, NAME, "no nodes are shutting down");
+    private static final Decision YES_NODE_NOT_SHUTTING_DOWN = Decision.single(Decision.Type.YES, NAME, "this node is not shutting down");
+
     /**
      * Determines if a shard can be allocated to a particular node, based on whether that node is shutting down or not.
      */
     @Override
     public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
-        final SingleNodeShutdownMetadata thisNodeShutdownMetadata = allocation.nodeShutdowns().get(node.nodeId());
+        final var nodeShutdowns = allocation.nodeShutdowns();
```

Comment:
It looks like this logic to decide which message is reported can be common and extracted into a validation method. It could simplify making a unit test too.

Reply:
Good point, in fact there's no real need to have different messages for the allocate and auto-expand cases either so the whole thing can just be a single method. See f034829a3b981114428d240cf1780d322a2c8f5d.